gridded_etl_tools/utils/extractor.py

```diff
@@ -116,6 +117,27 @@ def request(self, remote_file_path: str, local_file_path: str) -> bool:
             outfile.write(self.dm.session.get(remote_file_path).content)
         return True
 
+    def get_hrefs(self, url: str) -> list[str]:
+        """Get a list of links from a given URL
+
+        Parameters
+        ----------
+        url : str
+            A URL to scrape links from
+
+        Returns
+        -------
+        list[str]
+            A list of links, in string format, from the specified URL
+        """
+        self.dm.info("Getting links from: " + url)
+        page = self.dm.session.get(url, timeout=10)
+        html = page.content.decode("utf-8")
+        tree = etree.parse(StringIO(html), parser=etree.HTMLParser())
+        refs = tree.xpath("//a")
+        href_links = set([link.get("href", "") for link in refs])
+        return [link for link in href_links if "https" not in link and "mailto" not in link]
```

Comment:
Between CIDs and obscure scientific naming schemes it's actually possible these strings would appear in a link name... I think `not link.startswith("https://")` and `not link.startswith("mailto:")` would be OK but also I'm about to leave another comment about using BeautifulSoup.

Also excluding https from the results is confusing as default behavior for a library function